webapp/src/components/expanded_view/component.tsx

```diff
@@ -592,9 +594,16 @@ export default class ExpandedView extends React.PureComponent<Props, State> {
     public componentDidMount() {
         const callsClient = getCallsClient();
         if (!callsClient) {
+            logErr('callsClient should be defined');
             return;
         }
 
+        // On mount we request the call state through websocket. This avoids
+        // making a potentially racy HTTP call and should guarantee
+        // a consistent state.
+        logDebug('requesting call state through ws');
+        this.context.sendMessage('custom_com.mattermost.calls_call_state', {channelID: callsClient.channelID});
```

Comment:
I wonder if the websocket message isn't a bit confusing. Are you sending the call's state, or requesting it...?

Reply:
Just being consistent with other events there. We are using the direction (from/to client) to implicitly define whether it's request or response. I know you don't love it :p

Reply:
I guess HTTP fixes the confusion through the Method. Here we use a bit of context as it doesn't make any sense for the client to ever send the call state. Please let me know if this is blocking :)

Reply:
Nope, not blocking :)